mriqc/config.py

```diff
@@ -405,24 +408,49 @@ def init(cls):
         """Create a new BIDS Layout accessible with :attr:`~execution.layout`."""
         if cls._layout is None:
             import re
-
+            from bids.layout.index import BIDSLayoutIndexer
             from bids.layout import BIDSLayout
 
-            work_dir = cls.work_dir / "bids.db"
-            work_dir.mkdir(exist_ok=True, parents=True)
-            cls._layout = BIDSLayout(
-                str(cls.bids_dir),
+            _db_path = cls.bids_database_dir or (
+                cls.work_dir / cls.run_uuid / "bids_db"
+            )
+            _db_path.mkdir(exist_ok=True, parents=True)
+
+            # Recommended after PyBIDS 12.1
+            _indexer = BIDSLayoutIndexer(
                 validate=False,
-                # database_path=str(work_dir),
                 ignore=(
                     "code",
                     "stimuli",
                     "sourcedata",
                     "models",
                     "derivatives",
+                    "scripts",
                     re.compile(r"^\."),
+                    re.compile(
```

Comment:
why are all the below to be ignored (may be mriqc has no clue how to QC those yet)? might be handy to have a comment with the reasoning

Reply:
> may be mriqc has no clue how to QC those yet

correct



> may be it is possible with pybids to limit via inclusion not exclusion?

We are talking about that above https://github.com/nipreps/mriqc/pull/939#discussion_r717789263 - let's follow up there.

Reply:
Okay, I have just tried with MyConnectome the following:
```Python
indexer = BIDSLayoutIndexer(
    validate=False,
    ignore=("sourcedata", re.compile(r".*"), ),
    force_index=(
        re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/anat/.*_T(1|2)w\."),
        re.compile(r"sub-[a-zA-Z0-9]+(/ses-[a-zA-Z0-9]+)?/func/.*_bold\."),
    ),
)
```

and 1) sourcedata keeps being indexed (it matches the force_index); 2) it takes forever to initialize the layout, much more than the deny list approach.

btw inserting `^` in the begining makes it stop matching anything.